org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/AutoEditsPreferencePage.scala

```diff
@@ -65,7 +79,7 @@ class AutoEditsPreferencePage extends PreferencePage with IWorkbenchPreferencePa
       }
     }
     viewer.addCheckStateListener { e: CheckStateChangedEvent =>
-      toggleAutoEdit(e.getElement().asInstanceOf[AutoEditSetting])
+      prefStore.setValue(e.getElement.asInstanceOf[AutoEditSetting].id, e.getChecked)
```

Comment:
I don't see why you need to do this eagerly. Why not call `e.getChecked` on Apply or OK, for each checkbox?